internal/conf/load.go

```diff
@@ -0,0 +1,137 @@
+// INTECH Process Automation Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package conf
+
+import (
+	"fmt"
+	"gopkg.in/yaml.v3"
+	"io/ioutil"
+	"os"
+	"path"
+	"strconv"
+	"strings"
+)
+
+const (
+	Prefix    = "KUIPER"
+	Separator = "__"
+)
+
+func LoadConfig(c interface{}) error {
+	return LoadConfigByName(ConfFileName, c)
+}
+
+func LoadConfigByName(name string, c interface{}) error {
+	dir, err := GetConfLoc()
+	if err != nil {
+		return err
+	}
+	p := path.Join(dir, name)
+	return LoadConfigFromPath(p, c)
+}
+
+func LoadConfigFromPath(path string, c interface{}) error {
+	b, err := ioutil.ReadFile(path)
+	if err != nil {
+		return err
+	}
+	configMap := make(map[string]interface{})
+	err = yaml.Unmarshal(b, &configMap)
+	if err != nil {
+		return err
+	}
+	err = process(configMap, os.Environ())
+	if err != nil {
+		return err
+	}
+	var bs []byte
+	bs, err = yaml.Marshal(configMap)
```

Comment:
@ngjaying I am not sure how to make it better, I need to convert map[string]interface{} to struct / interface

Reply:
@ngjaying in case of camelCase. I have not handled it properly - I will need to add mapping but I have feeling that proper hierarchy in they yaml config would solve the problem

Reply:
ah ok. I think current approach is ok. It is just an one time action. Does this also solve the problem of lowercase -> camelCase?

Reply:
nope, I am about commit fix about camel case handling,I have changed Unmatshalling/Marshalling to sth that looks better at a cost of additional dependency.

Reply:
I have commited it right now, I have feeling that it looks better right now